Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -1832,7 +1987,11 @@ RedistributeMPI (std::map<int, Vector<char> >& not_ours,
             for (int i = 0; i < int(Cnt); ++i)
             {
                 char* pbuf = ((char*) &recvdata[offset]) + i*superparticle_size;
-                ParticleType p;
+                auto& ptile = m_particles[rcv_levs[ipart]][std::make_pair(rcv_grid[ipart],
+                                                                          rcv_tile[ipart])];
+
+                auto p = make_particle<ParticleType>{}(ptile.getParticleTileData(),ipart);
+                
                 std::memcpy(&p, pbuf, sizeof(ParticleType));
                 locateParticle(p, pld, lev_min, lev_max, nGrow);
```

Comment:
@atmyers can you please help with the logic in these lines?
Should we write a for loop instead of the memcpy here for the pure SoA particle?

Maybe we can also check how the structure in `recvdata` is looking now for SoA particles, is this still in AoS?

Reply:
I believe all the actual data payload is stored in the SoA components, correct? If so, I think the data will be redistributed properly in the logic for the SoA components below. We just need to make sure to use `0` for the size of these memcpy calls and when we increment the pointer. 

The SoA data are transposed to SoA and back again when copying to / from the mpi buffers. We could potentially look at optimizing that for pure SoA case, but correctness first.

Reply:
We can do this:
```C++
if constexpr(PureSoA) {
    AMREX_D_TERM(
        p.pos(0) = static_cast<amrex::ParticleReal*>(pbuf)[0];,
        p.pos(1) = static_cast<amrex::ParticleReal*>(pbuf)[1];,
        p.pos(2) = static_cast<amrex::ParticleReal*>(pbuf)[2];
   )
}
else {
    std::memcpy(&p, pbuf, sizeof(ParticleType));
}
```
I don't think that we need to handle `id` and `cpu` here, right? :)
(in that case, the first branch of this `if` even works for both data layouts.)